Applications/Spire/Source/Ui/PopupBox.cpp

```diff
@@ -0,0 +1,182 @@
+#include "Spire/Ui/PopupBox.hpp"
+#include <QApplication>
+#include "Spire/Ui/Layouts.hpp"
+
+using namespace boost::signals2;
+using namespace Spire;
+
+PopupBox::PopupBox(QWidget& body, QWidget* parent)
+    : QWidget(parent),
+      m_body(&body),
+      m_window(nullptr),
+      m_focus_observer(*this),
+      m_body_focus_observer(*m_body),
+      m_position_observer(*this),
+      m_alignment(Alignment::NONE),
+      m_min_height(m_body->sizeHint().height()),
+      m_max_height(QWIDGETSIZE_MAX),
+      m_above_space(0),
+      m_below_space(0),
+      m_right_space(0),
+      m_focus_connection(m_focus_observer.connect_state_signal(
```

Comment:
There is no need for an object to establish a connection to another object that is owned. Since ```PopupBox``` owns ```m_focus_observer``` then no connection is needed since the ```m_focus_observer``` will be destroyed before the ```PopupBox``` is destroyed.